### ISSUE TITLE
macOS .app bundle busted on Mojave

### ISSUE THREAD

**Issue by floam** [fish-shell/fish-shell#5727]
Double clicking fish.app does nothing on 10.14 for me.

I notice this stuff in through `log stream`:

```
133: 0xa1cd5	0	0	appinstallation	11:40:54.889494 -0800	Parsed Info.plist for com.ridiculousfish.fish-shell	LaunchServices	com.apple.launchservices	lsd
133: 0xa1cd5	0	0	appinstallation	11:40:54.889548 -0800	Parsed MobileInstallation data for com.ridiculousfish.fish-shell	LaunchServices	com.apple.launchservices	lsd
294: 0xa1b13	0	382676	access	11:40:55.107756 -0800	target_executable_path_URL: file:///var/folders/0q/8cc45czd1ps_vbn96jn87xhc0000gn/T/AppTranslocation/28201531-34A6-465E-BEA6-A59ABA9263ED/d/fish-4.app/Contents/MacOS/fish_launcher	tccd	com.apple.TCC	tccd
294: 0xa1b13	1	382676	access	11:40:55.113632 -0800	-[TCCDAccessIdentity initWithIdentifier:type:executableURL:SDKVersion:]: self.bundle=0x7fb2c6f0e7a0, bundle:NSBundle </var/folders/0q/8cc45czd1ps_vbn96jn87xhc0000gn/T/AppTranslocation/28201531-34A6-465E-BEA6-A59ABA9263ED/d/fish-4.app> (not yet loaded); for: com.ridiculousfish.fish-shell, URL: file:///var/folders/0q/8cc45czd1ps_vbn96jn87xhc0000gn/T/AppTranslocation/28201531-34A6-465E-BEA6-A59ABA9263ED/d/fish-4.app/Contents/MacOS/fish_launcher, /var/folders/0q/8cc45czd1ps_vbn96jn87xhc0000gn/T/AppTranslocation/28201531-34A6-465E-BEA6-A59ABA9263ED/d/fish-4.app/Contents/MacOS/fish_launcher	tccd	com.apple.TCC	tccd
294: 0xa1b13	1	382676	access	11:40:55.113943 -0800	do_TCCAccessRequestIndirect_block_invoke: indirect object Requirements match for com.ridiculousfish.fish-shell : com.apple.Terminal	tccd	com.apple.TCC	tccd
294: 0xa1b13	1	382676	access	11:40:55.114097 -0800	Handling access request: kTCCServiceAppleEvents:com.ridiculousfish.fish-shell:com.apple.Terminal, default_allow: 0, allowed: 0, prompt_count:0	tccd	com.apple.TCC	tccd
294: 0xa1b13	1	382676	access	11:40:55.114884 -0800	No usage string found (key:NSAppleEventsUsageDescription) for client[0] in bundle:/var/folders/0q/8cc45czd1ps_vbn96jn87xhc0000gn/T/AppTranslocation/28201531-34A6-465E-BEA6-A59ABA9263ED/d/fish-4.app	tccd	com.apple.TCC	tccd
294: 0xa1b13	0	382676	access	11:40:55.115052 -0800	Refusing TCCAccessRequestIndirect for service kTCCServiceAppleEvents and client /var/folders/0q/8cc45czd1ps_vbn96jn87xhc0000gn/T/AppTranslocation/28201531-34A6-465E-BEA6-A59ABA9263ED/d/fish-4.app[0] without NSAppleEventsUsageDescription key	tccd	com.apple.TCC	tccd
50946: 0xa1ded	1	0	receive	11:40:55.120900 -0800	REPLY:(aevt,ansr) {aevt,ansr target=Terminal {errn=-1743} returnID=-1867}	AE	com.apple.appleevents	fish_launcher
50946: 0xa1ded	0	0		11:40:55.152111 -0800	Oops: {
    NSAppleScriptErrorAppName = Terminal;
    NSAppleScriptErrorBriefMessage = "Not authorized to send Apple events to Terminal.";
    NSAppleScriptErrorMessage = "Not authorized to send Apple events to Terminal.";
    NSAppleScriptErrorNumber = "-1743";
    NSAppleScriptErrorRange = "NSRange: {0, 0}";
}	fish_launcher		fish_launcher
78: 0x23e	0	0	default	11:40:55.152717 -0800	fish_launcher [50946]:77367 -> diskarbitrationd [78]	diskarbitrationd	com.apple.DiskArbitration.diskarbitrationd	diskarbitrationd
78: 0x23e	0	0	default	11:40:55.152784 -0800	  removed session, id = fish_launcher [50946]:77367.	diskarbitrationd	com.apple.DiskArbitration.diskarbitrationd	diskarbitrationd
85: 0xa1d64	1	0	session	11:40:55.155401 -0800	PID: 50946, Client: 'fish_launcher', exited with 0 session(s), 0 node(s) and 0 active request(s)	opendirectoryd	com.apple.opendirectoryd	opendirectoryd
```

**Comment by floam** [comment]
This line in the output explains why:

`Refusing TCCAccessRequestIndirect for service kTCCServiceAppleEvents and client /var/folders/0q/8cc45czd1ps_vbn96jn87xhc0000gn/T/AppTranslocation/28201531-34A6-465E-BEA6-A59ABA9263ED/d/fish-4.app[0] without NSAppleEventsUsageDescription key`

We need a `NSAppleEventsUsageDescription` key in the Info.plist that explains why we want to send Apple Events. The key is required for Mojave if the app is built against the 10.14 SDK.

**Comment by floam** [comment]
See https://www.felix-schwarz.org/blog/2018/08/new-apple-event-apis-in-macos-mojave

**Comment by floam** [comment]
Fixed in that commit, but I am reopening because the .app bundle on fishshell.com is still busted. I think @ridiculousfish should just cherry-pick that fix and quietly re-sign the 3.0.2 .app bundle so it's not broken for people.

**Comment by ridiculousfish** [comment]
Done, sorry for delay :/
